Networking/Networking/Network/AlamofireNetwork.swift

```diff
@@ -79,38 +82,44 @@ public class AlamofireNetwork: Network {
     /// - Parameter request: Request that should be performed.
     /// - Returns: A publisher that emits the result of the given request.
     public func responseDataPublisher(for request: URLRequestConvertible) -> AnyPublisher<Swift.Result<Data, Error>, Never> {
-        let request = createRequest(wrapping: request)
-
         return Future() { promise in
-            Alamofire.request(request).responseData { response in
-                let result = response.result.toSwiftResult()
-                promise(Swift.Result.success(result))
+            self.requestAuthenticator.authenticateRequest(request) { request in
+                Alamofire.request(request).responseData { response in
+                    let result = response.result.toSwiftResult()
+                    promise(Swift.Result.success(result))
+                }
             }
         }.eraseToAnyPublisher()
     }
 
     public func uploadMultipartFormData(multipartFormData: @escaping (MultipartFormData) -> Void,
                                         to request: URLRequestConvertible,
                                         completion: @escaping (Data?, Error?) -> Void) {
-        let request = createRequest(wrapping: request)
-
-        backgroundSessionManager.upload(multipartFormData: multipartFormData, with: request) { (encodingResult) in
-            switch encodingResult {
-            case .success(let upload, _, _):
-                upload.responseData { response in
-                    completion(response.value, response.error)
+        requestAuthenticator.authenticateRequest(request) { [weak self] request in
+            guard let self else { return }
+            self.backgroundSessionManager.upload(multipartFormData: multipartFormData, with: request) { (encodingResult) in
+                switch encodingResult {
+                case .success(let upload, _, _):
+                    upload.responseData { response in
+                        completion(response.value, response.error)
+                    }
+                case .failure(let error):
+                    completion(nil, error)
                 }
-            case .failure(let error):
-                completion(nil, error)
             }
         }
     }
 }
 
-private extension AlamofireNetwork {
-    func createRequest(wrapping request: URLRequestConvertible) -> URLRequestConvertible {
-        credentials.map { AuthenticatedRequest(credentials: $0, request: request) } ??
-        UnauthenticatedRequest(request: request)
+public extension AlamofireNetwork {
+    /// Updates the application password use case with a new site ID.
+    ///
+    func configureApplicationPasswordHandler(with siteID: Int64) {
+        guard let credentials else {
+            return
+        }
+        let applicationPasswordUseCase = TemporaryApplicationPasswordUseCase(siteID: siteID, credentials: credentials)
+        requestAuthenticator.updateApplicationPasswordHandler(with: applicationPasswordUseCase)
```

Comment:
Yup, that was what I followed initially - but then I found that injecting the use case makes  `RequestAuthenticator` more testable. Otherwise, I'm not sure how to test it with a mock use case. Please let me know if you have any suggestions regarding testability without an injected use case.

Reply:
Oh, that is a good point. 

Can we use `MockNetwork` to mock the responses for unit testing? i.e. Inject a network into `RequestAuthenticator` and use that to initialize `ApplicationPasswordUseCase`